juju/client/connector.py

```diff
@@ -104,12 +104,19 @@ def connection(self):
         except JujuError as e:
             raise JujuConnectionError(e.message) from e
         if controller is None:
-            raise JujuConnectionError('Controller {} not found'.format(controller_name))
+            raise JujuConnectionError('Controller {} not found'.format(
+                controller_name))
         # TODO change Connection so we can pass all the endpoints
         # instead of just the first one.
         endpoint = controller['api-endpoints'][0]
-        models = self.jujudata.models()[controller_name]
-        account = self.jujudata.accounts()[controller_name]
+        accounts = self.jujudata.accounts()
+        if controller_name not in accounts:
+            raise JujuAuthError('Account not found for controller: {}'.format(
+                controller_name))
+        account = accounts[controller_name]
+        models = self.jujudata.models().get(controller_name, {})['models']
```

Comment:
This looks a bit odd - if `self.jujudata.models()` doesn't container `controller_name`, then it'll default to `{}`, try to get `'models'` from it, and raise a `KeyError` yeah?